modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/SortedIndexDescriptor.java

```diff
@@ -0,0 +1,206 @@
+
+        /**
+         * Returns {@code true} if this column is sorted in ascending order or {@code false} otherwise.
+         */
+        public boolean asc() {
+            return asc;
+        }
+
+        /**
+         * Returns {@code true} if this column was explicitly marked as an indexed column or {@code false} if it is a part of a Primary Key
+         * appended for uniqueness.
+         */
+        public boolean indexedColumn() {
+            return indexedColumn;
+        }
+
+        /**
+         * Returns {@code true} if this column can contain null values or {@code false} otherwise.
+         */
+        public boolean nullable() {
+            return nullable;
+        }
+
+        @Override
+        public String toString() {
+            return S.toString(this);
+        }
+    }
+
+    private final String name;
+
+    private final List<ColumnDescriptor> columns;
+
+    private final SchemaDescriptor schemaDescriptor;
+
+    /**
+     * Creates an Index Descriptor from a given Table Configuration.
+     *
+     * @param name        index name.
+     * @param tableConfig table configuration.
+     */
+    public SortedIndexDescriptor(String name, TableView tableConfig) {
+        this.name = name;
+
+        TableIndexView indexConfig = tableConfig.indices().get(name);
+
+        if (indexConfig == null) {
+            throw new StorageException(String.format("Index configuration for \"%s\" could not be found", name));
+        }
+
+        if (!(indexConfig instanceof SortedIndexView)) {
+            throw new StorageException(String.format(
+                    "Index \"%s\" is not configured as a Sorted Index. Actual type: %s",
+                    name, indexConfig.type()
+            ));
+        }
+
+        // extract indexed column configurations from the table configuration
+        NamedListView<? extends IndexColumnView> indexColumns = ((SortedIndexView) indexConfig).columns();
+
+        Stream<String> indexColumnNames = indexColumns.namedListKeys().stream();
+
+        // append the primary key to guarantee index key uniqueness
+        Stream<String> primaryKeyColumnNames = Arrays.stream(tableConfig.primaryKey().columns());
+
+        List<ColumnView> indexKeyColumnViews = Stream.concat(indexColumnNames, primaryKeyColumnNames)
+                .distinct() // remove Primary Key columns if they are already present in the index definition
+                .map(columnName -> {
+                    ColumnView columnView = tableConfig.columns().get(columnName);
+
+                    assert columnView != null : "Incorrect index column configuration. " + columnName + " column does not exist";
+
+                    return columnView;
+                })
+                .collect(toList());
+
+        schemaDescriptor = createSchemaDescriptor(indexKeyColumnViews);
+
+        columns = Arrays.stream(schemaDescriptor.keyColumns().columns())
+                .sorted(Comparator.comparingInt(Column::columnOrder))
```

Comment:
Will this be the same?
```suggestion
        columns = indexKeyColumnViews.stream()
```

Reply:
Unfortunately, no. `columns` returned from `indexKeyColumnViews` don't have a `schemaIndex` =(

Reply:
But you use columnOrder().
Its an order the column are in the index descriptor... seems, you just restore the indexKeyColumnViews order here.

Reply:
yes, but I later need the correct `schemaIndex` inside the `BinaryRowComparator` for example. So, the only problem is that `indexKeyColumnViews` do not contain fully initialized `Column` objects, that are later needed.